reporting/reporting.go

```diff
@@ -0,0 +1,90 @@
+package reporting
+
+import "fmt"
+
+var startReportFormat = "⌛ %s...%s"
+var continueReportFormat = "\r⌛ %s...%s"
+var completeReportFormat = "\r✅  %s\n"
+var failReportFormat = "\r❌ %s\n"
+var successfullListing = "👍  %s\n"
+var failedListing = "👎  %s\n%s\n"
+
+func getFormatString(processes []process) string {
+	if len(processes) > 0 {
+		return continueReportFormat
+	}
+
+	return startReportFormat
+}
+
+type process struct {
+	name      string
+	succeeded bool
+	errorMsg  string
+}
+
+func newProcess(name string) process {
+	return process{name: name}
+}
+
+type Reporter struct {
+	processes     []process
+	longestString int
+}
+
+func NewReporter() Reporter {
+	return Reporter{}
+}
+
+func (r *Reporter) Report(process string) {
+	padding := r.updateProcessLengthAndGetPadding(process)
+	fmt.Printf(getFormatString(r.processes), process, padding)
+	r.processes = append(r.processes, newProcess(process))
+}
+
+func (r *Reporter) updateProcessLengthAndGetPadding(process string) string {
+	var padding string
+	if len(process) > r.longestString {
+		r.longestString = len(process)
+	} else {
+		for i := len(process); i <= r.longestString; i++ {
+			padding = padding + " "
+		}
```

Comment:
Seems like this for loop is all you need. You don't need the if check above. If `len(process)` is > `r.longestString`, this loop will run zero times.

Reply:
the `if` is how i'm updating `longestString`. Without it, how do you suggest updating `longestString`?

Reply:
Oh whoops. Missed that.